hue/hue.sh

```diff
@@ -24,127 +29,142 @@ function update_apt_get() {
   return 1
 }
 
-# Determine the role of this node
-ROLE=$(/usr/share/google/get_metadata_value attributes/dataproc-role)
-
-# Only run on the master node of the cluster
-if [[ "${ROLE}" != 'Master' ]]; then
-  exit 0
-fi
-
+function install_hue_and_configure(){
+local old_hdfs_url='## webhdfs_url\=http:\/\/localhost:50070'
+local new_hdfs_url='webhdfs_url\=http:\/\/'"$(hdfs getconf -confKey  dfs.namenode.http-address)"
```

Comment:
Since we're cleaning things up anyway, can you just do `cat ${new_hdfs_url} >> /path/to/ini` (and similarly for mysql). As long as Hue always picks up the last thing in the ini file, we can ignore whatever existed before.

Actually, you could combine the three of these (including line 81) into `cat << EOF >> ini-file ... EOF`, and it will be a lot more readable.

Reply:
Much more readable – I agree.  I also tested this change and it seems to be working the same way. This file is based on high number of lines and sections. Is it ok to put this line after this:
```
###########################################################################
# Settings to configure Metadata
###########################################################################

[metadata]
  

  [[optimizer]]

  [[navigator]]
 

  hadoop_mapred_home=/usr/lib/hadoop-mapreduce
```

Reply:
@karth295 What do you think about it? In my opinion using ```cat << EOF >> ini-file ... EOF``` could give a duplicated lines inside ini-file. It can obsfuscate ini file.